proposals/lip-0045.md

```diff
@@ -984,164 +839,223 @@ The Interoperability module contains the following commands:
+The `data` property of this event uses the `chainAccountSchema` schema given [above](#json-schema-1).
+
+#### CrossChainMessageProcessed
+
+This event has `typeID = TYPE_ID_CCM_PROCESSED`. This event is emitted when a CCM is sent or processed.
+The status property of the data schema can take one of the following values:
 
-##### Parameters
+1. `CCM_PROCESSED_STATUS_SENT`: The CCM was sent to the receiving chain.
+2. `CCM_PROCESSED_STATUS_CHANNEL_UNAVAILABLE`: The receiving chain is not available.
+3. `CCM_PROCESSED_STATUS_MODULE_NOT_SUPPORTED`: The receiving chain does not support the module.
+4. `CCM_PROCESSED_STATUS_COMMAND_NOT_SUPPORTED`: The receiving chain does not support the command.
+5. `CCM_PROCESSED_STATUS_FORWARDED`: The CCM was forwarded to the receiving chain.
+6. `CCM_PROCESSED_STATUS_APPLIED`: The CCM was applied on the receiving chain.
+7. `CCM_PROCESSED_STATUS_DISCARDED`: The CCM was discarded.
+8. `CCM_PROCESSED_STATUS_RECOVERY_FORWARDED`: The CCM was forwarded to the original sending chain for recovery.
+9. `CCM_PROCESSED_STATUS_RECOVERY_APPLIED`: The CCM was recovered and applied on the original sending chain.
+10. `CCM_PROCESSED_STATUS_RECOVERY_DISCARDED`: The CCM was ricovered and discarded.
 
-This function has the following input parameters in the order given below:
 
-* `ccm`: The cross-chain message.
-* `errorStatus`: The error status to be set.
+##### Topics
 
-##### Returns
+- `sendingChainID`: The ID of the sending chain.
+- `receivingChainID`: The ID of the receiving chain.
+- `ccmID`: The ID of the CCM that was sent. This topic is present only for newly sent messages (event status equals `CCM_PROCESSED_STATUS_SENT`).
 
-A boolean indicating the success of the function execution.
+Furthermore, one
 
-##### Execution
+##### Data
+
+```java
+ccmProcessedDataSchema = {
+    "type": "object",
+    "required": ["ccmID", "status"],
+    "properties": {
+        "ccmID": {
+            "dataType": "bytes",
+            "length": HASH_LENGTH, 
+            "fieldNumber": 1
+        },
+        "status": {
+            "dataType": "uint32",
+            "fieldNumber": 2
+        }
+    }
+}
+```
+
+### Protocol Logic for Other Modules
+
+#### send
+
+The `send` function is used to create and add a message to the outbox of a partner chain.
 
 ```python
-error(ccm, errorStatus):
-    # Error codes from 0 to MAX_RESERVED_ERROR_STATUS (included) are reserved to the Interoperability module
-    if 0 <= errorStatus <= MAX_RESERVED_ERROR_STATUS:
-        return False
+def send(sendingAddress: Address, 
+    moduleID: ModuleID, 
+    crossChainCommandID: CrossChainCommandID, 
+    receivingChainID: ChainID, 
+    fee: uint64, 
+    status: uint32, 
+    params: object) -> None:
+    # Processing on the mainchain
+    if ownChainAccount.ID == MAINCHAIN_ID:
+        # Chain has to exist on mainchain
+        if chainAccount(receivingChainID) does not exists:
+            raise Exception('Receiving chain does not exist.')
```

Comment:
Why did you decide not to have an error event in this case?

Reply:
Makes sense to emit event actually. I moved the ccm sent event to a new event and added errors.